### PR TITLE
fixes for: Replay Multi-session, Mitm Socket, Proxy Errors

### PR DESCRIPTION
**Pull request by blakebyrnes**
This PR fixes 3 bugs:
1 - handle named sessions in replay (and don't overwrite the currently open script instance when multiple run at once)
2 - #98 properly bubble up proxy errors (and improve naming)
3 - mitm incorrectly tracking proxy connection back to itself, which means resources in concurrent sessions can get improperly assigned to other sessions

chore: require await when using async via eslint rule. Without this rule, we are seeing promise chains explode before reaching the client.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -172,6 +184,10 @@ module.exports = {
     '@typescript-eslint/no-empty-interface': 'off',
     '@typescript-eslint/no-namespace': 'off',
     '@typescript-eslint/ordered-imports': 'off',
+    '@typescript-eslint/no-shadow': [
+      'error',
+      { ignoreTypeValueShadow: true, ignoreFunctionTypeParameterNameValueShadow: true },
+    ],
```
Comment (calebjclark) [inline .eslintrc.js:190]: Why?

File: .github/workflows/lint-and-test.yml [hunk 2]
```diff
@@ -116,7 +116,7 @@ jobs:
           SA_REPLAY_SKIP_BINARY_DOWNLOAD: 1
 
       - name: Run tests
-        run: yarn jest --testTimeout=15000 --runInBand
+        run: yarn jest --testTimeout=60000 --runInBand
```
Comment (calebjclark) [inline .github/workflows/lint-and-test.yml:119]: That's a major increase.

File: .prettierrc [hunk 3]
```diff
@@ -2,5 +2,6 @@
   "printWidth": 100,
   "singleQuote": true,
   "trailingComma": "all",
-  "tabWidth": 2
+  "tabWidth": 2,
+  "arrowParens": "avoid"
```
Comment (calebjclark) [inline .prettierrc:6]: Why? https://prettier.io/docs/en/options.html#arrow-function-parentheses

Comment (blakebyrnes) [reply .prettierrc:6]: This was causing changes to the code base as it stands currently everytime I was hitting save and auto-formatting. I chose to change the rule to match the current code

File: .github/workflows/lint-and-test.yml (hunk 2 above)
Comment (blakebyrnes) [reply .github/workflows/lint-and-test.yml:119]: It was an attempt to workaround these sporadically failing tests that happens almost exclusively on windows. I didn't spend the time to figure out how to change for only windows.

File: .eslintrc.js (hunk 1 above)
Comment (blakebyrnes) [reply .eslintrc.js:190]: Cause you love shadows?!?! I would have just made it a straight error, but ... you love shadows :)

Comment (blakebyrnes) [reply .eslintrc.js:190]: I was trying to find the equivalent rule to our old rule about shadows being ok when they don't change the outer scope, but this was the closest I could get with the @typescript-eslint module

**Pull request merged**